pandas/tests/io/formats/test_format.py

```diff
@@ -689,6 +689,31 @@ def test_to_string_with_formatters(self):
         )
         assert result == result2
 
+    def test_to_string_with_truncated_formatters(self):
+        df = DataFrame(
+            {
+                "int": [1, 2, 3],
+                "float": [1.0, 2.0, 3.0],
+                "object": [(1, 2), True, False],
+            },
+            columns=["int", "float", "object"],
+        )
+
+        formatters = [
+            ("int", lambda x: f"[1] {x}"),
+            ("float", lambda x: f"[2] {x}"),
+            ("object", lambda x: f"[3] {x}"),
+        ]
+        result = df.to_string(formatters=dict(formatters), max_cols=2)
```

Comment:
Rather than this can you just parametrize the inputs?

Reply:
I'm not sure what you mean.

Reply:
Check out the `pytest.mark.parametrize` decorator - it is used by a few other tests in this module already

Reply:
I haven't used that before, but I'm still not sure what you had in mind to parameterize this method.

Reply:
I added parameterized inputs.  I'm not sure if this is quite what you were expecting since there was only one pair of inputs / outputs used in the test.

Reply:
I would argue that the original version (without parametrize) was more readable. Maybe @WillAyd suggested that you parameterize separately formatters for float, int and object?

Reply:
That wouldn't exercise the fix though.  It needs all of the columns in one DataFrame so that a truncated string representation can be displayed.

Reply:
The suggestion was to parametrize the arguments to `formatters`; one parametrize for the dict input and one for the list